core/src/main/java/org/hisp/dhis/android/core/settings/internal/LatestAppVersionCall.kt

```diff
@@ -46,22 +46,27 @@ internal class LatestAppVersionCall(
 
     override suspend fun tryFetch(storeError: Boolean): Result<LatestAppVersion, D2Error> {
         return coroutineAPICallExecutor.wrap(storeError = storeError) {
-            val userGroupUids = userModule.userGroups().blockingGetUids()
+            val version = resolveApkDistributionVersion()
 
-            val versions = settingAppService.versions().items()
+            version?.let { LatestAppVersion.builder().version(it.version).downloadURL(it.downloadURL).build() }
+                ?: settingAppService.latestAppVersion()
+        }
+    }
 
-            val filteredVersions = versions.filter { version ->
-                version.userGroups?.any { userGroupUid ->
-                    userGroupUids.contains(userGroupUid)
-                } ?: false
-            }
+    internal suspend fun resolveApkDistributionVersion(): ApkDistributionVersion? {
+        val userGroupUids = userModule.userGroups().blockingGetUids()
 
-            val version = filteredVersions.maxWithOrNull(versionComparator.comparator)
-                ?: versions.find { it.isDefault == true }
+        val versions = settingAppService.versions().items()
 
-            version?.let { LatestAppVersion.builder().version(it.version).downloadURL(it.downloadURL).build() }
-                ?: settingAppService.latestAppVersion()
+        val filteredVersions = versions.filter { version ->
+            version.userGroups?.any { userGroupUid ->
+                userGroupUids.contains(userGroupUid)
+            } ?: false
         }
+
+        val availableVersions = filteredVersions + versions.find { it.isDefault == true }!!
```

Comment:
The configuration should include a version with the property `isDefault` set to true. If there is no default version, this line will throw a NPE because of the `!!`.

I will try to make the method more tolerant to unexpected configurations. For example:
- if there is no default version but the user has any group with an assigned version, use that version.
- if there are more than one default versions, take the highest one (instead of the first one).

If we wanted to validate that there is just one default version, that is valid too. In that case I will throw an explicit `InvalidArgumentException` with a message instead of a generic NPE caused by `!!`.